cmd/upgrade_test.go

```diff
@@ -4,104 +4,169 @@
-			}
+	for _, c := range cases {
+		err := c.uc.validate(r)
+		if c.expectedErr != nil && err != nil {
+			g.Expect(err.Error()).To(Equal(c.expectedErr.Error()))
+		} else {
+			g.Expect(err).To(BeNil())
+			g.Expect(c.expectedErr).To(BeNil())
 		}
+	}
+}
+
+func TestCreateUpgradeCommandSuccesfully(t *testing.T) {
+	g := NewGomegaWithT(t)
+	output := newUpgradeCmd()
+
+	g.Expect(output.Use).Should(Equal(upgradeName))
+	g.Expect(output.Short).Should(Equal(upgradeShortDescription))
+	g.Expect(output.Long).Should(Equal(upgradeLongDescription))
+	g.Expect(output.Flags().Lookup("location")).NotTo(BeNil())
+	g.Expect(output.Flags().Lookup("resource-group")).NotTo(BeNil())
+	g.Expect(output.Flags().Lookup("deployment-dir")).NotTo(BeNil())
+	g.Expect(output.Flags().Lookup("upgrade-version")).NotTo(BeNil())
+}
+
+//TODO: Should it fail or should it pass without --force? hmm
+func TestUpgradeShouldFailForSameVersion(t *testing.T) {
+	fakeARMTemplateHandle := strings.NewReader(`{"parameters" : { "nameSuffix" : {"defaultValue" : "test"}}}`)
+	g := NewGomegaWithT(t)
+	upgradeCmd := &upgradeCmd{
+		resourceGroupName:   "rg",
+		deploymentDirectory: "_output/test",
+		upgradeVersion:      "1.10.13",
+		location:            "centralus",
+		timeoutInMinutes:    60,
+
+		client: &armhelpers.MockAKSEngineClient{},
+	}
+
+	containerServiceMock := api.CreateMockContainerService("testcluster", "1.10.13", 3, 2, false)
+	containerServiceMock.Location = "centralus"
+	upgradeCmd.containerService = containerServiceMock
+	err := upgradeCmd.validateCurrentLocalState(fakeARMTemplateHandle)
+	g.Expect(err).To(HaveOccurred())
+	g.Expect(err.Error()).To(ContainSubstring("upgrading from Kubernetes version 1.10.13 to version 1.10.13 is not supported"))
+}
+
+func TestUpgradeFailWithPathWhenAzureDeployJsonIsInvalid(t *testing.T) {
+	fakeARMTemplateHandle := strings.NewReader(`{"parameters" : { "nameSuffix"}}`)
+	g := NewGomegaWithT(t)
+	upgradeCmd := &upgradeCmd{
+		resourceGroupName:   "rg",
+		deploymentDirectory: "_output/myspecialtestfolder",
+		upgradeVersion:      "1.13.3",
+		location:            "centralus",
+		timeoutInMinutes:    60,
+		force:               true,
+		client:              &armhelpers.MockAKSEngineClient{},
+	}
+
+	containerServiceMock := api.CreateMockContainerService("testcluster", "1.13.2", 3, 2, false)
+	containerServiceMock.Location = "centralus"
+	upgradeCmd.containerService = containerServiceMock
+	err := upgradeCmd.validateCurrentLocalState(fakeARMTemplateHandle)
+	g.Expect(err).To(HaveOccurred())
+	g.Expect(err.Error()).To(ContainSubstring("_output/myspecialtestfolder/azuredeploy.json"))
+}
+func TestUpgradeForceSameVersionShouldSucceed(t *testing.T) {
+	fakeARMTemplateHandle := strings.NewReader(`{"parameters" : { "nameSuffix" : {"defaultValue" : "test"}}}`)
+	g := NewGomegaWithT(t)
+	upgradeCmd := &upgradeCmd{
+		resourceGroupName:   "rg",
+		deploymentDirectory: "_output/test",
+		upgradeVersion:      "1.10.13",
```

Comment:
Does this test (and any other tests) require this k8s patch version to be supported? If so you could use `common.GetLatestPatchVersion` to get a valid 1.10 version and avoid having to update this UT every time we deprecate a patch version.